packages/analytics-js/src/components/userSessionManager/UserSessionManager.ts

```diff
@@ -349,16 +417,30 @@ class UserSessionManager implements IUserSessionManager {
     state.session.initialReferringDomain.value = referrer;
   }
 
-  // TODO: session tracking
-  startAutoTracking() {}
+  /**
+   * A function to check for existing session details and depending on that create a new session.
+   */
+  startAutoTracking() {
+    const timestamp = Date.now();
+    if (!isValidSession(timestamp)) {
+      state.session.sessionInfo.value = generateAutoTrackingSession(timestamp);
+    }
+  }
 
-  // TODO: session tracking
-  start(sessionId?: number) {}
+  /**
+   * A function method to start a manual session
+   * @param {number} sessionId     session identifier
+   * @returns
+   */
+  start(id?: number) {
+    state.session.sessionInfo.value = generateManualTrackingSession(id, this.logger);
+  }
 
-  // TODO: session tracking
+  /**
+   * A public method to end an ongoing session.
+   */
   end() {
-    this.reset(false, true);
-    this.clearUserSessionStorage();
+    state.session.sessionInfo.value = {};
```

Comment:
should we not clear storage when session ends?

Reply:
It will be done automatically through effect